### PR TITLE
Curvenote Start

### PR DESCRIPTION
**Pull request by rowanc1**
* Superseeds #44

### CODE REVIEW

File: src/cli/services/utils/utils.ts [hunk 1]
```diff
@@ -1,28 +1,70 @@
...
+${chalk.bold('https://nodejs.org/en/download/')}
+
+You can upgrade your NPM version using:
+
+${chalk.bold('npm install -g npm@latest')}
+`;
+
+async function checkNodeVersion(session: ISession): Promise<boolean> {
+  const checking = new Promise<boolean>((resolve) => {
+    check({ node: '>= 14.0.0 <17.0.0', npm: '>7' }, (error, result) => {
```
Comment (rowanc1) [inline src/cli/services/utils/utils.ts:22]: @Yxwww thoughts on what this should be?

File: src/sync/add.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      ]);
+    }
+    firstTime = false;
+  }
+  return config;
+}
+
+const START = `
+
+${chalk.bold(chalk.green('Add new content to your Curvenote project'))} 🛻
```
Comment (rowanc1) [inline src/sync/add.ts:85]: Can chain these.

File: src/web/roles.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  'r:dynamic': RDynamic,
+  'r:display': RDisplay,
+  'r:range': RRange,
+  cite: Cite,
+  'cite:p': CiteP,
+  cite_group: CiteP,
+  'cite:t': CiteT,
+  u: Underline,
+  underline: Underline,
+};
```
Comment (rowanc1) [inline src/web/roles.ts:176]: Very much looking forward to some better processors/extensions here.

File: src/web/transforms/keys.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  return node;
+}
+
+/**
+ * Add unique keys to every node
+ *
+ * @param mdast
+ * @returns
+ */
+export function transformKeys<T extends GenericNode | Root>(mdast: T) {
```
Comment (rowanc1) [inline src/web/transforms/keys.ts:16]: @fwkoch this is where the transforKeys is.

**Review by rowanc1 — COMMENTED** [review]
A few comments.

**Comment by rowanc1** [comment]
@fwkoch I am putting some pieces in now that might have to change once we get some better controls on MyST JS.

I think one of the requests I would have is to be able to have fine-grained control over the heading numbering.

image

File: src/web/transforms/images.ts [hunk 5]
```diff
@@ -58,41 +80,19 @@ export async function transformImages(mdast: Root, state: TransformState) {
         // Assume non-oxa, non-url paths are local images relative to the config.section.path
         const fullPath = path.join(state.folder, image.url);
         if (!fs.existsSync(fullPath)) {
-          session.log.debug(`Cannot find image: ${fullPath}`);
+          session.log.error(`Cannot find image: ${fullPath}`);
```
Comment (fwkoch) [inline src/web/transforms/images.ts:83]: No errors on external / oxa images, though? Makes some sense, since the user has more control over local images.

Comment (rowanc1) [reply src/web/transforms/images.ts:83]: Will cover the other ones as well!

**Pull request merged**